src/system.jl

```diff
@@ -249,11 +262,11 @@ Base.@kwdef struct GeneratorTimeSeries
     "Generation of the generator at the start of the time period (pu)"
     initial_generation::KeyedArray{Float64, 1}
     "Generator offer curves. `KeyedArray` where the axis keys are `generator names x datetimes`"
-    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}, 2}
+    offer_curve::KeyedArray{PriceSensitiveBid, 2}
     "Generator minimum output in the ancillary services market (pu)"
-    regulation_min::KeyedArray{Float64, 2}
+    regulation_min::Union{Missing, KeyedArray{Float64, 2}}
     "Generator maximum output in the ancillary services market (pu)"
-    regulation_max::KeyedArray{Float64, 2}
+    regulation_max::Union{Missing, KeyedArray{Float64, 2}}
```

Comment:
If we intend to reuse code, is it easier to set the values to something like `Inf` rather than checking for missings?

Reply:
Interesting question, [this is how](https://gitlab.invenia.ca/invenia/research/FullNetworkModels.jl/-/blob/19a7be6ccc41b770d93fb6554313b15dfa13a104/src/model/general/constraints.jl#L855) `regulation_max` is used in FNModels.  I think if we set it to `Inf` and tried to define that constraint, that wouldn't give us the intended behaviour?

Reply:
I'm not sure but it looks like it would work out to a constraint that is always true? We might have to reformat the expression or add an extra constraint but it might be worth it to avoid having to special-case things to account for possible missings.